src/world/action/walk.c

```diff
@@ -1,13 +1,125 @@
 
 // walk
 INCLUDE_ASM(void, "world/action/walk", func_802B6000_E236E0, void);
 
 // run
-INCLUDE_ASM(void, "world/action/walk", func_802B6288_E23968, void);
+void func_802B6288_E23968() {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PlayerData* playerData = &gPlayerData;
+    f32 sp14;
+    f32 sp10;
+    s32 temp_v1;
+    s32 phi_a0;
+    f32 phi_f2;
+    s32 phi_s3;
+    s32 xAxis;
+    s32 yAxis;
+
+    phi_s3 = 0;
+    if (playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS) {
+        func_802B6738_E23E18();
+        return;
+    }
+
+    if (playerStatus->flags < 0) {
+        playerStatus->flags &= ~0x80880000;
+        D_8010C980 = 0;
+        playerStatus->unk_60 = 0;
+        phi_s3 = 1;
+
+        if (!(playerStatus->flags & 0x00004000)) {
+            playerStatus->currentSpeed = playerStatus->runSpeed;
+        }
+        if (playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO) {
+            phi_a0 = 0x90003;
+        } else {
+            if (!(playerStatus->animFlags & PLAYER_ANIM_FLAG_HOLDING_ITEM)) {
+                phi_a0 = 0x10005;
+            } else {
+                phi_a0 = 0x60002;
+            }
+        }
+        suggest_player_anim_clearUnkFlag(phi_a0);
+    }
+
+    if (playerStatus->flags & 0x00004000) {
+        playerStatus->targetYaw = playerStatus->heading;
+        try_player_footstep_sounds(4);
+        return;
+    }
 
-// same as func_802B6198_E24768 but requires data stuff
-INCLUDE_ASM(void, "world/action/walk", func_802B6550_E23C30, void);
+    D_8010C980++;
+    phi_f2 = 1.0f;
+
+    if (playerStatus->animFlags & 0x10000) {
+        phi_f2 = 1.5f;
+    }
+
+    playerStatus->currentSpeed = playerStatus->runSpeed * phi_f2;
+    player_input_to_move_vector(&sp10, &sp14);
+    phys_update_interact_collider();
+    if (check_input_jump() == FALSE) {
+        if (phi_s3 == 0) {
+            if (check_input_hammer()) {
+                return;
+            }
+        }
+
+        player_input_to_move_vector(&sp10, &sp14);
+        if (sp14 == 0.0f) {
+            set_action_state(ACTION_STATE_IDLE);
+            return;
+        }
+
+        if (fabsf(D_800F7B40 - sp10) <= 90.0f) {
+            temp_v1 = playerStatus->animFlags;
+            if (temp_v1 >= 0) {
+                playerStatus->targetYaw = sp10;
+            }
+           playerStatus->animFlags &= ~0x80000000;
+        } else {
+            temp_v1 = playerStatus->animFlags;
+            if (temp_v1 < 0) {
+                playerStatus->targetYaw = sp10;
+            } else {
+                xAxis = 0x80000000;
+                playerStatus->animFlags |= xAxis;
+            }
+
+        }
+
+        if (!is_ability_active(0xB)) {
+            xAxis = playerStatus->stickAxis[0] * playerStatus->stickAxis[0];
+            yAxis = playerStatus->stickAxis[1] * playerStatus->stickAxis[1];
+            if (55.0f >= sqrtf(xAxis + yAxis)) {
```

Comment:
Yuck, a yoda condition. Does it match this way?
```suggestion
            if (sqrtf(xAxis + yAxis) <= 55.0f) {
```

Reply:
Looking at line 207 of this file, this should be able to match without the xAxis and yAxis temps. Could you also use the SQ (square) macro?